src/System.Linq.Expressions/src/System/Linq/Expressions/IndexExpression.cs

```diff
@@ -472,7 +472,7 @@ private static void ValidateAccessorArgumentTypes(MethodInfo method, ParameterIn
 
                     Type pType = pi.ParameterType;
                     if (pType.IsByRef) throw Error.AccessorsCannotHaveByRefArgs($"{nameof(indexes)}[{i}]");
-                    TypeUtils.ValidateType(pType, $"{nameof(indexes)}[{i}]");
+                    TypeUtils.ValidateType(pType, nameof(indexes));
```

Comment:
It seems we're using the index in only a few places to enhance the exception message. I agree it provides useful data, so I'll go through the code and find places where the index is not incorporated and add it. My thinking is to push down the pair of `paramName` and `index` to the `Error` class and centralize the `$"{paramName}[{index}]" formatting there.